### ISSUE TITLE
clang-tidy modernize-avoid-c-arrays on std::unique_ptr<T[]>

### ISSUE THREAD

**Issue by avlec** [llvm/llvm-project#107913]
I don't think this should warn?

Seeing on clang-tidy 17.0.6 also tested on trunk on godbolt running clang-tidy tool with the argument `--checks=modernize-avoid-c-arrays`

```cpp
#include <memory>

auto func(int num) {
    return std::unique_ptr<char[]> {};
}
```

**Comment by chrchr-github** [comment]
Duplicate of https://github.com/llvm/llvm-project/issues/40517
